core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -105,6 +107,14 @@ private[spark] class CoarseGrainedExecutorBackend(
       executor.stop()
       context.stop(self)
       context.system.shutdown()
+
+    // Add new credentials received from the driver to the current user.
+    case UpdateCredentials(newCredentials) =>
```

Comment:
Would it be feasible to write the new credentials to HDFS and just send their location in this message? That way the security of this feature doesn't depend on securing the Spark transport layer.